[Ace] Adds a feature fpic to enable -fPIC compilation on Linux

pierrebizz opened the pull request:
Adds a feature fpic to enable -fPIC compilation on Linux in order to use the ACE/TAO libraries in shared libs.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <all>, <No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

Commit:
support of -fPIC flag in linux compilation

Commit:
support of -fPIC flag in linux compilation

Commit:
support -fPIC flag in compilation

Commit:
support -fPIC flag in compilation

Commit:
Update ace.json

JonLiu1993 commented on ports/ace/portfile.cmake:
```diff
@@ -93,7 +93,7 @@ if(VCPKG_TARGET_IS_WINDOWS)
   elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
     set(SOLUTION_TYPE vs2017)
-  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
+  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142") 
```
```suggestion
  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
```

JonLiu1993 replied:
@pierrebizz ,Thanks for your pr, Please remove this extra space

pierrebizz replied:
Done

RockinRoel committed:
[wt] update to 4.6.1 (#22772)

* [wt] update to 4.6.1

Setting WTHTTP_CONFIGURATION and CONFIGURATION to empty string as well,
because these would otherwise be filled in with an absolute path.
Setting them to empty string ensures they won't be used.

* update version

Co-authored-by: Jonliu1993 <13720414433@163.com>

JackBoosY committed:
[libheif] Fix usage (#22818)

* [libheif] Fix usage

* version

* Add license

* version

ras0219-msft committed:
[ci.baseline.txt] Remove fails for passing ports (#22821)

abique committed:
[alsa] update to 1.2.6.1 (#22824)

pierrebizz committed:
[ace] build windows (#22832)

* Fix Build Windows - include files not copied correctly to target

* Update versions

* install_includes requires quotes on windows

* Update ace.json

Co-authored-by: bisaillonp <Pierre.Bisaillon@ixiasoft.com>

autoantwort committed:
[detours,devicenameresolver,dlfcn-win32,duilib,winpcap] add supports expression (#22776)

autoantwort committed:
[ogre] no absolute paths (#22354)

dg0yt committed:
[vcpkg-cmake] Remove cmake generator support restrictions (#22626)

* Stop failing in build step on unknown generators

CMake already completed configuration using the given generator.

* Support parallel build with Unix Makefiles

* Documentation fixes for vcpkg_cmake port

* `vcpkg_cmake_configure`: links to `opencv4` as example instead of
  `opencv` which is an empty port
* `vcpkg_cmake_install`: fix link to `vcpkg_cmake_build`'s docs
* `vcpkg_cmake_build`: fix link to `vcpkg_cmake_install`'s docs

* Extra script audit

* Update versions

* Regenerate docs

* [vcpkg-cmake] support Xcode parallel build

* no/parallel_args when CMake Generator is  "Xcode"

* Update versions

* CI

Co-authored-by: Martin Hořeňovský <martin.horenovsky@gmail.com>
Co-authored-by: luncliff <dong-ha-park@linecorp.com>

autoantwort committed:
[vcpkg] collect error logs in logs.txt in the buildtrees folder (#22146)

* collect error logs in logs.txt in the buildtrees folder

* Use global variable

* rename file to error-logs-.txt and remove error: prefix in file

pierrebizz committed:
Rebase on master/Update version

pierrebizz committed:
Modify versions

pierrebizz committed:
Merge branch 'ace-build-linux' of https://github.com/pierrebizz/vcpkg into ace-build-linux

pierrebizz committed:
Superfluous space

JonLiu1993 commented:
@pierrebizz ,Please check your commit branch, it looks like you introduced a lot of unnecessary changes

pierrebizz commented:
I will redo the Pull Request; I kind of rebased master instead of merging it ; I'm not familiar with GitHub workflow.
